### ISSUE TITLE
Support for specifying memory and CPU resources on SGE

### ISSUE THREAD

**Issue by bow** [broadinstitute/cromwell#583]
Currently (HEAD points to 284ba81 at the time of writing) there is no way to specify memory and CPU resources when jobs are run using the SGE backend. I think this would be one of the critical features to have a production-ready SGE backend implementation.

**Comment by delocalizer** [comment]
I suggest also allow specifying walltime and queue; walltime in particular as it enables most efficient performance of the scheduler and is often a required resource request in traditional HPC cluster setups.

**Comment by ffinfo** [comment]
I can confirm that it is possible to set CPU's, memory, queue and project in the current develop branch. Only wall time I'm not so sure.

@kshakir: Maybe you know more about this?

**Comment by kshakir** [comment]
Fixed in #1252.

"walltime" may now read as a backend specific runtime attribute just like any other. I've called the runtime attribute "sge_walltime" below.

```
SGE {
  actor-factory = "cromwell.backend.impl.sfs.config.ConfigBackendLifecycleActorFactory"

  config {

    runtime-attributes = """
    Int cpu = 1
    Float? memory_gb
    String? sge_queue
    String? sge_project
    String? sge_walltime
    """

    submit = """
    qsub \
    -terse \
    -V \
    -b n \
    -N ${job_name} \
    -wd ${cwd} \
    -o ${out} \
    -e ${err} \
    -pe smp ${cpu} \
    ${"-l m_mem_free=" + memory_gb + "gb"} \
    ${"-q " + sge_queue} \
    ${"-P " + sge_project} \
    ${"-l h rt=" + sge_walltime } \
    ${script}
    """

    kill = "qdel ${job_id}"

    check-alive = "qstat -j ${job_id}"

    job-id-regex = "(\\d+)"

  }
}
```
